app/controllers/api/conversion_hosts_controller.rb

```diff
@@ -1,5 +1,58 @@
 module Api
   class ConversionHostsController < BaseController
     include Subcollections::Tags
+
+    # Create a conversion host and enable it. If you specify a 'queue'
+    # parameter with a true value, the enablement configuration will run as
+    # an MiqTask. Both the 'resource_type' and 'resource_id' are mandatory
+    # arguments, and the 'resource_type' must be either 'Host' or 'VmOrTemplate'.
+    #
+    # You may optionally pass in 'param_v2v_vddk_package_url' or 'auth_user'
+    # arguments as well. The 'auth_user' is only used if the 'queue' parameter
+    # is true.
+    #
+    # POST /api/conversion_hosts {
+    #   "name": "some_name",
+    #   "queue": "true",
+    #   "resource_type": "Host",
+    #   "resource_id": 7
+    #   "param_v2v_vddk_package_url": "some_url"
+    #   "auth_user": "some_user"
+    # }
+    #
+    def create_resource(_type, _id, data)
+      raise BadRequestError, "resource_id must be specified" unless data['resource_id']
+      raise BadRequestError, "resource_type must be specified" unless data['resource_type']
+
+      # Both the 'queue' and 'auth_user' params must be deleted since the model will
+      # otherwise pass the data hash directly as params to ConversionHost.new.
+      queue = data.delete('queue')
+
+      if queue
+        auth_user = data.delete('auth_user')
+        ConversionHost.enable_queue(data, auth_user)
+      else
+        ConversionHost.enable(data)
+      end
+    end
+
+    # Disable the conversion host role by installing the conversion host module
+    # and running the conversion host playbook that disables it. If you pass
+    # a true value to the 'queue' parameter, this method will use an MiqTask.
+    #
+    # POST /api/conversion_hosts/:id { "action": "disable" }
+    # POST /api/conversion_hosts/:id { "action": "disable", "queue": true }
+    # POST /api/conversion_hosts/:id { "action": "disable", "queue": true, "auth_user": "someone" }
+    #
+    def disable_resource(type, id, data)
+      conversion_host = resource_search(id, type, collection_class(type))
+      if data['queue'].present?
+        conversion_host.disable_queue(data['auth_user'])
+      else
+        conversion_host.disable
```

Comment:
similar to above

Reply:
Ok, I'll switch it so that `disable_queue` is the default.